src/vscode-dts/vscode.d.ts

```diff
@@ -19363,25 +19384,26 @@ declare module 'vscode' {
 		 * @param content The content of the message.
 		 * @param name The optional name of a user for the message.
 		 */
-		static User(content: string, name?: string): LanguageModelChatMessage;
+		static User(content: string | (LanguageModelTextPart | LanguageModelToolResultPart | LanguageModelToolCallPart)[], name?: string): LanguageModelChatMessage;
 
 		/**
 		 * Utility to create a new assistant message.
 		 *
 		 * @param content The content of the message.
 		 * @param name The optional name of a user for the message.
 		 */
-		static Assistant(content: string, name?: string): LanguageModelChatMessage;
+		static Assistant(content: string | (LanguageModelTextPart | LanguageModelToolResultPart | LanguageModelToolCallPart)[], name?: string): LanguageModelChatMessage;
 
 		/**
 		 * The role of this message.
 		 */
 		role: LanguageModelChatMessageRole;
 
 		/**
-		 * The content of this message.
+		 * A string or heterogeneous array of things that a message can contain as content. Some parts may be message-type
+		 * specific for some models.
 		 */
-		content: string;
+		content: string | (LanguageModelTextPart | LanguageModelToolResultPart | LanguageModelToolCallPart)[];
 
```

Comment:
Could we change `content` (the property) to be always and just an array of these parts? I believe there is no reading part for this in the API?